### PR TITLE
Next

### PR DESCRIPTION
**Pull request by jamarob**
rework routes and some bits here and there

### CODE REVIEW

File: api/todos/[id].js [hunk 1]
```diff
@@ -1,27 +1,53 @@
...
-    }
+        if(method === "PUT") {
+            const todoToUpdate = await Todo.findById(id)
+            if(!todoToUpdate) {
+                return response.status(404).send()
+            }
+            todoToUpdate.description = request.body.description
+            todoToUpdate.done = request.body.done
+            const updatedTodo = await todoToUpdate.save()
+            return response.status(200).json(updatedTodo)
```
Comment (macfrei) [inline api/todos/[id].js:27]: Why use `findById` and `save` here instead of `findByIdAndUpdate`?

File: api/todos/[id].js [hunk 2]
```diff
@@ -1,27 +1,53 @@
-import connectToMongodb from '../../backend/db/connect-to-mongodb'
-import Todo from '../../backend/model/Todo'
+import connectToMongodb from '../../src/api/db/connect-to-mongodb'
+import Todo from '../../src/api/model/Todo'
 
-const handler = async (request, response) => {
-  await connectToMongodb()
+const enclosedHandler = async (request, response) => {
```
Comment (macfrei) [inline api/todos/[id].js:4]: What does `enclosedHandler` mean?

File: api/todos/index.js [hunk 3]
```diff
@@ -1,24 +1,27 @@
-import connectToMongodb from '../../backend/db/connect-to-mongodb'
-import Todo from '../../backend/model/Todo'
+import connectToMongodb from '../../src/api/db/connect-to-mongodb'
+import Todo from '../../src/api/model/Todo'
 
-const handler = async (request, response) => {
-  await connectToMongodb()
+const enclosingHandler = async (request, response) => {
```
Comment (macfrei) [inline api/todos/index.js:4]: Why `enclosingHandler` here?

File: src/api/model/Todo.js [hunk 4]
```diff
@@ -3,8 +3,8 @@ const { Schema, model } = mongoose
```
Comment (macfrei) [inline src/api/model/Todo.js:3]: Why are the `db` and `model` file in the `src` folder?

**Review by macfrei — COMMENTED** [review]
Looks good, just a few questions :)

File: api/todos/[id].js (hunk 1 above)
Comment (jamarob) [reply api/todos/[id].js:27]: We validate at the data layer with mongoose and `findByIdAndUpdate` allows for partial updates. To be true to HTTP PUT semantics we need it to run validation for all parameters (not sending description **and** done fails).

File: api/todos/[id].js (hunk 2 above)
Comment (jamarob) [reply api/todos/[id].js:4]: `/books` is the enclosing resource and `/books/some-book-id` the enclosed resource. Think collection and and an item of that collection. But the name doesn't really matter here, vercel only cares about the default export.

File: api/todos/index.js (hunk 3 above)
Comment (jamarob) [reply api/todos/index.js:4]: see above

File: src/api/model/Todo.js (hunk 4 above)
Comment (jamarob) [reply src/api/model/Todo.js:3]: They are source files :) I think it is confusing to have a src folder and then another folder at the same level which holds source files

**Comment by jamarob** [comment]
> Looks good, just a few questions :)

Thank you :)

File: src/api/model/Todo.js (hunk 4 above)
Comment (macfrei) [reply src/api/model/Todo.js:3]: True. In my head it was more like: okay, we have the `src` folder with all the frontend stuff and then we have everything related to backend outside of the folder. 😅

Comment (jamarob) [reply src/api/model/Todo.js:3]: And I agree that would be a better folder structure (and not have the api folder outside of src as well), but then we need to configure the build and start scripts.. imho not worth the hassle 😅

**Pull request merged**